viam-cartographer.go

```diff
@@ -244,6 +246,23 @@ func (cartoSvc *cartographerService) GetInternalState(ctx context.Context) (func
 	return grpchelper.GetInternalStateCallback(ctx, cartoSvc.Name().ShortName(), cartoSvc.clientAlgo)
 }
 
+// GetLatestMapInfo forwards the request for timestamp data to the slam library's gRPC service.
+// Once a response is received, it is converted into a time.Time object.
```

Comment:
[major] The way this function is currently written, it will forward this call to the slam algorithm's grpc server (in this case this server is being run by the c++ binary). However, for GetLatestMapInfo, we want to instead directly return the time recorded in GetPointCloudMap.